src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,600 @@
+
+    if (!this->isOutputValid())
+    {
+        log()->warn("{} Returning an empty output.", logPrefix);
+        return m_pImpl->outputRef;
+    }
+
+    m_pImpl->outputRef = *m_pImpl->output;
+    return m_pImpl->outputRef;
+}
+
+bool Planners::UnicyclePlanner::isOutputValid() const
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::isOutputValid]";
+
+    if (!m_pImpl->planner)
+    {
+        log()->error("{} The Unicycle planner has never been initialized.", logPrefix);
+        return false;
+    }
+
+    if (!m_pImpl->left || !m_pImpl->right)
+    {
+        log()->error("{} The Unicycle planner never computed the foot steps.", logPrefix);
+        return false;
+    }
+
+    if (!m_pImpl->output)
+    {
+        log()->error("{} The output has never been computed.", logPrefix);
+        return false;
+    }
+
+    return true;
+}
+
+bool Planners::UnicyclePlanner::setInput(const UnicyclePlannerInput& input)
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::setInput]";
+
+    if (!m_pImpl->planner)
+    {
+        log()->error("{} The Unicycle planner has never been initialized.", logPrefix);
+        return false;
+    }
+
+    auto getMaxKnotTime = [](const UnicyclePlannerInput& input) -> double {
+        double maxKnotTime = 0.0;
+
+        for (const auto& knot : input.knots)
+        {
+            if (knot.time > maxKnotTime)
+                maxKnotTime = knot.time;
+        }
+
+        return maxKnotTime;
+    };
+
+    if (input.tf < getMaxKnotTime(input))
+    {
+        log()->error("{} The input contains a knot whose time is over the planner horizon.",
+                     logPrefix);
+        return false;
+    }
+
+    m_pImpl->planner->clearDesiredTrajectory();
+
+    for (const auto& knot : input.knots)
+    {
+        auto position = iDynTree::Vector2();
+        position[0] = knot.x;
+        position[1] = knot.y;
+
+        auto velocity = iDynTree::Vector2();
+        velocity[0] = knot.dx;
+        velocity[1] = knot.dy;
+
+        if (!m_pImpl->planner->addDesiredTrajectoryPoint(knot.time, position, velocity))
+        {
+            m_pImpl->planner->clearDesiredTrajectory();
+            log()->error("{} Failed to insert knot in the Unicycle planner.", logPrefix);
+            return false;
+        }
+    }
+
+    m_pImpl->horizon.t0 = input.t0;
+    m_pImpl->horizon.tf = input.tf;
+
+    return true;
+}
+
+bool Planners::UnicyclePlanner::advance()
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::advance]";
+
+    if (!m_pImpl->planner)
+    {
+        log()->error("{} The Unicycle planner has never been initialized.", logPrefix);
+        return false;
+    }
+
+    m_pImpl->output = std::nullopt;
+
+    // Initialize the left FootPrint
+    m_pImpl->left = std::make_shared<FootPrint>();
+    m_pImpl->left->setFootName(m_pImpl->names.left);
+
+    // Initialize the right FootPrint
+    m_pImpl->right = std::make_shared<FootPrint>();
+    m_pImpl->right->setFootName(m_pImpl->names.right);
```

Comment:
https://github.com/robotology/unicycle-footstep-planner/blob/d3f6c80afe21a9958da769c8dd8a2bbfee5ea922/src/UnicycleGenerator.cpp#L395-L409 in case you don't need to change the position or the orientation of the planned footsteps.

Reply:
> This means that you are always expecting the trajectory to start with the feet parallel and centered in the origin. Is this wanted?

In this initial implementation, this was the testing scenario. I'll try adapting it with a custom initial configuration.